tests/unit/test_charm.py

```diff
@@ -915,3 +915,214 @@ def test_postgres_patch_storage_config_defaults_to_database_relation(self):
+            )
+
+            # Adding another unit of `pro-airgapped-server`, but this new unit should not
+            # affect the Livepatch server configuration.
+            self.harness.add_relation_unit(pro_rel_id, "pro-airgapped-server/1")
+            self.harness.update_relation_data(
+                pro_rel_id,
+                "pro-airgapped-server/1",
+                {
+                    "scheme": "scheme",
+                    "hostname": "second.host",
+                    "port": "9999",
+                },
+            )
+
+            self._assert_environment_contains(
+                {
+                    "LP_CONTRACTS_ENABLED": True,
+                    "LP_CONTRACTS_URL": "scheme://first.host:9999",
+                }
+            )
+
+            # Now we drop remove the first `pro-airgapped-server` unit. The charm should
+            # use the second unit address.
+            self.harness.remove_relation_unit(pro_rel_id, "pro-airgapped-server/0")
+
+            self._assert_environment_contains(
+                {
+                    "LP_CONTRACTS_ENABLED": True,
+                    "LP_CONTRACTS_URL": "scheme://second.host:9999",
+                }
+            )
+
+    def test_pro_airgapped_server_relation__relation_removed(self):
+        """Test when pro-airgapped-server is removed."""
+        self.harness.set_leader(True)
+        self.harness.enable_hooks()
+
+        self.harness.charm._state.dsn = "postgresql://123"
+        self.harness.charm._state.resource_token = TEST_TOKEN
+
+        with patch("src.charm.LivepatchCharm.migration_is_required") as migration:
+            migration.return_value = False
+            self.harness.update_config(
+                {
+                    "server.url-template": "http://localhost/{filename}",
+                    "server.is-hosted": False,
+                }
+            )
+
+            pro_rel_id = self.harness.add_relation("pro-airgapped-server", "pro-airgapped-server")
+            self.harness.add_relation_unit(pro_rel_id, "pro-airgapped-server/0")
+            self.harness.update_relation_data(
+                pro_rel_id,
+                "pro-airgapped-server/0",
+                {
+                    "scheme": "scheme",
+                    "hostname": "first.host",
+                    "port": "9999",
+                },
+            )
+
+            self._assert_environment_contains(
+                {
+                    "LP_CONTRACTS_ENABLED": True,
+                    "LP_CONTRACTS_URL": "scheme://first.host:9999",
+                }
+            )
+
+            # Now we remove the relation. The plan should now point to the hosted contracts service.
+            self.harness.remove_relation(pro_rel_id)
+
+            self._assert_environment_contains(
+                {
+                    "LP_CONTRACTS_ENABLED": True,
+                    "LP_CONTRACTS_URL": "https://contracts.canonical.com",
+                }
+            )
```

Comment:
Why should it point to hosted instead of being empty?

Reply:
This is the default one, so how it should be.